sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -261,8 +272,43 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     // it makes sense to execute them after drawing a map but before swapping buffers
     // **note** this queue also holds snapshot tasks
     drainQueue(renderEventQueue)
+    // calculate FPS here, before actual swap as swap could happen either this or next frame
+    val actualEndRenderTimeNanos = SystemClock.elapsedRealtimeNanos()
+    if (renderTimeNsCopy != 0L && actualEndRenderTimeNanos < expectedEndRenderTimeNs) {
+      // we need to stop swap buffers for less than time requested in order to have some time to render upcoming frame
+      // before next vsync so it will be drawn, otherwise we will drop it
+      expectedVsyncWakeTimeNs = expectedEndRenderTimeNs - ONE_MILLISECOND_NS
+    }
+    fpsChangedListener?.let {
+      val fps = 1E9 / (actualEndRenderTimeNanos - timeElapsed)
+      if (timeElapsed != 0L) {
+        it.onFpsChanged(fps)
+      }
+      timeElapsed = actualEndRenderTimeNanos
+    }
+    if (needViewAnnotationSync && viewAnnotationMode == ViewAnnotationUpdateMode.MAP_SYNCHRONIZED) {
```

Comment:
wdyt about introducing boolean property that checks both of them, and naming it like `swapBuffersOnNextVsync` or similar

Reply:
see no sense as it's checked in a single place and is more easy at least for me to understand then one more extra bool flag